packages/ra-core/src/core/useResourceDefinition.ts

```diff
@@ -6,9 +6,34 @@ import { useResourceContext } from './useResourceContext';
 /**
  * Hook which returns the definition of the requested resource
  */
-export const useResourceDefinition = (props): ResourceDefinition => {
+export const useResourceDefinition = (
+    props: UseResourceDefinitionOptions
+): ResourceDefinition => {
     const resource = useResourceContext(props);
     const resources = useSelector(getResources);
 
+    if (
+        props.resource &&
+        (typeof props.hasCreate !== 'undefined' ||
+            typeof props.hasEdit !== 'undefined' ||
+            typeof props.hasList !== 'undefined' ||
+            typeof props.hasShow !== 'undefined')
+    ) {
+        return {
+            ...props,
```

Comment:
You should whitelist the props that can fit the resource definition type, otherwise the hook may return an incorrect object. Check at #5802 for an example.

Reply:
Props are typed to at least implement the required interface, isn't that enough?

Reply:
at compile time, yes, at runtime, no.